File: basics/Q29/main.go
package main

import (
	"fmt"
	"os/exec"
	"strings"
)

func main() {
	cmd := exec.Command("go", "list", "all")
	out, err := cmd.Output()
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	packages := strings.Split(string(out), "\n")

	for _, pkg := range packages {
		if pkg != "" {
			fmt.Println(pkg)
		}
	}
}

File: basics/Q29/main.rs
use std::process::Command;

fn main() {
    let output = Command::new("go").args(["list", "all"]).output();

    let output = match output {
        Ok(o) => o,
        Err(e) => {
            println!("Error: {}", e);
            return;
        }
    };

    if !output.status.success() {
        let err_text = String::from_utf8_lossy(&output.stderr);
        let msg = err_text.trim();
        if msg.is_empty() {
            println!("Error: command failed");
        } else {
            println!("Error: {}", msg);
        }
        return;
    }

    let out_text = String::from_utf8_lossy(&output.stdout);
    for line in out_text.split('\n') {
        if !line.is_empty() {
            println!("{}", line);
        }
    }
}